components/sup/src/manager/service/mod.rs

```diff
@@ -926,10 +986,46 @@ impl Service {
                 (false, _) => HealthCheck::Critical,
             }
         };
+        if check_result == HealthCheck::Ok {
+            self.schedule_routine_health_check();
+            debug!(
+                "Service ({}) health check is: {}",
+                self.spec_ident, check_result
+            );
+        } else {
+            self.schedule_special_health_check();
+            debug!(
+                "Service ({}) health check is: {}. Re-running within default {:#?}",
+                self.spec_ident,
+                check_result,
+                *HealthCheckInterval::default().as_ref()
+            );
+        }
+        self.health_check = check_result;
         self.last_health_check = Some(Instant::now());
+        self.special_health_check_scheduled = false;
         self.cache_health_check(check_result);
     }
 
+    fn schedule_routine_health_check(&mut self) {
+        let interval = self.health_check_interval.clone();
```

Comment:
```rust
        self.schedule_health_check(self.health_check_interval);
```
won't work because of:
```
error[E0503]: cannot use `self.health_check_interval` because it was mutably borrowed                                                                                                                       
    --> components/sup/src/manager/service/mod.rs:1031:36                                                                                                                                                   
     |                                                                                                                                                                                                      
1031 |         self.schedule_health_check(self.health_check_interval);                                                                                                                                      
     |         ----                       ^^^^^^^^^^^^^^^^^^^^^^^^^^ use of borrowed `*self`                                                                                                                
     |         |                                                                                                                                                                                            
     |         borrow of `*self` occurs here                                                                                                                                                                
```
But since `HealthCheckInteval` is `Copy` type, there's not need to `clone`:
```suggestion
        let interval = self.health_check_interval;
```

Reply:
It's really odd that it compiles as-is for me 🤔 but I agree with your comments here and am taking your suggestion also yields a successful compile.

Reply:
It's not surprising that it compiles (everything that's `Copy` also implements `Clone`), but since `clone` typically implies we're doing something expensive, avoiding calling it on a `Copy` type helps prevent unnecessary head-scratching.

Reply:
Thank you for the clarification - makes perfect sense.